esp/bindings/SOAP/xpp/sxt/XmlTokenizer.h

```diff
@@ -201,6 +201,59 @@ namespace sxt {
     paramNoMixContent = !enable;
   }
 
+  // brorrow from JLIB
+  // Return: the # of bytes
+  static const unsigned UTF8_MAX_BYTES = 6;
+  static unsigned writeUtf8(unsigned c, unsigned char* outBuf)
+  {
+    int idx = 0;
+    if (c < 0x80)
+    {
+      outBuf[idx++] = (unsigned char)c;
+    }
+    else if (c < 0x800)
+    {
+      outBuf[idx++] =  (unsigned char)(0xC0 | c >> 6);
+      outBuf[idx++] = (unsigned char)(0x80 | c & 0x3F);
+    }
+    else if (c < 0x10000)
+    {
+      outBuf[idx++] = (unsigned char)(0xE0 | c >> 12);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 6 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c & 0x3F);
+    }
+    else if (c < 0x200000)
+    {
+      outBuf[idx++] = (unsigned char)(0xF0 | c >> 18);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 12 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 6 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c & 0x3F);
+    }
+    else if (c < 0x4000000)
+    {
+      outBuf[idx++] = (unsigned char)(0xF8 | c >> 24);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 18 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 12 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 6 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c & 0x3F);
+    }
+    else if (c < 0x80000000)
+    {
+      outBuf[idx++] = (unsigned char)(0xFC | c >> 30);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 24 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 18 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 12 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c >> 6 & 0x3F);
+      outBuf[idx++] = (unsigned char)(0x80 | c & 0x3F);
+    }
+    else {
+      char buf[256];
+      snprintf(buf, 256, "Internal error at %s(%d) in %s", __FILE__, __LINE__, __func__);
```

Comment:
Could provide a bit more description.  Such as "invalid UCS character" or "out of bounds UCS character"...
 at %s(%d) in %s.